CRM/Price/BAO/PriceFieldValue.php

```diff
@@ -125,6 +124,16 @@ public static function create(&$params, $ids = array()) {
         }
       }
     }
+    if ($id) {
+      $financialType = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceFieldValue', $id, 'financial_type_id', 'id');
+    }
+    else {
+      $financialType = CRM_Utils_Array::value('financial_type_id', $params, NULL);
+    }
+    CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($financialTypes);
```

Comment:
There is actually a validateFinancialTypes function - perhaps this should go inside that?